galite-core/src/main/kotlin/org/kopi/galite/visual/Item.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2013-2020 kopiLeft Services SARL, Tunis TN
+ * Copyright (c) 1990-2020 kopiRight Managed Solutions GmbH, Wien AT
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1 as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package org.kopi.galite.visual
+
+import org.kopi.galite.base.Image
+
+class Item(var id: Int,
+           var parent: Int,
+           var name: String,
+           var localisedName: String?,
+           var description: String?,
+           var selected: Boolean,
+           var defaultItem: Boolean,
+           icon: String?,
+           var originalName: String) : Comparable<Item> {
+
+  lateinit var icon: Image
+  lateinit var smallIcon: Image
+  var childCount = 0
+  lateinit var children: Array<Item>
+  var level = 0
+
+  init {
+    this.id = id
+    this.parent = parent
+    this.name = name
+    this.localisedName = localisedName
+    this.description = description
+    this.selected = selected
+    this.defaultItem = defaultItem
+    this.originalName = originalName
+    if (icon != null) {
+      this.icon = ImageHandler.imageHandler.getImage(icon)
+      smallIcon = ImageHandler.imageHandler.getImage(icon)
+      if (smallIcon == null) {
+        smallIcon = smallIcon?.getScaledInstance(16, 16, Image.SCALE_SMOOTH)
+      }
+    }
+  }
```

Comment:
check if we can change this : 
`class Item(val id: Int,
           var parent: Int,
           var name: String,
           var localisedName: String?,
           var description: String?,
           var selected: Boolean,
           var defaultItem: Boolean,
           icon: String?,
           var originalName: String) : Comparable<Item> {

  lateinit var icon: Image
  lateinit var smallIcon: Image
  var childCount = 0
  lateinit var children: Array<Item>
  var level = 0

  init {
    if (icon != null) {
      this.icon = ImageHandler.imageHandler.getImage(icon)
      smallIcon = ImageHandler.imageHandler.getImage(icon)
      if (smallIcon == null) {
        smallIcon = smallIcon?.getScaledInstance(16, 16, Image.SCALE_SMOOTH)
      }
    }
  }`

Reply:
Changed , only for icon and smallIcon ( changed to nullable and you can't put lateinit  with nullable types )